ui/pages/send/send-content/send-content.component.js

```diff
@@ -66,6 +70,10 @@ export default class SendContent extends Component {
       asset.type !== ASSET_TYPES.TOKEN &&
       asset.type !== ASSET_TYPES.COLLECTIBLE;
 
+    if (!this.state.isKnownContractAddress) {
```

Comment:
This should be done in a `componentDidUpdate` call. Ideally, the render method does not have state management side effects. Also, with the `componentDidUpdate` method, we can compare `prevProps.recipient` to `this.props.recipient` and only call `this.checkContractAddress` when they are different. This ensures we don't call it more than necessary, and that we do call it if the recipient changes.

Reply:
Fixed in: e3b7ee44fa5b49dbce8251dcb856212b1b34eb85